app/scripts/metamask-controller.js

```diff
@@ -3526,6 +3577,20 @@ export default class MetamaskController extends EventEmitter {
       });
     };
     this.on('update', handleUpdate);
+    const startUISync = () => {
+      if (outStream._writableState.ended) {
+        return;
+      }
+      // send notification to client-side
+      outStream.write({
+        jsonrpc: '2.0',
+        method: 'startUISync',
+      });
+    };
+    this.on('startUISync', startUISync);
+    if (this.startUISync) {
+      this.emit('startUISync');
+    }
```

Comment:
Nit: Rather than setting up a listener just to call it once, we can call the function directly

```suggestion
    if (this.startUISync) {
      startUISync()
    } else {
       this.on('startUISync', startUISync);
    }
```